Sources/NIO/ByteBuffer-views.swift

```diff
@@ -0,0 +1,51 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+public struct ByteBufferView: ContiguousCollection {
+    public typealias Element = UInt8
+    public typealias Index = Int
+
+    private let buffer: ByteBuffer
+
+    internal init(buffer: ByteBuffer, range: Range<Int>) {
```

Comment:
Is the range actually a good type here? I'm inclined to say that start index and length is more likely to be helpful here.

Reply:
@Lukasa but range is just a start index and an end index?

Reply:
Sure, but it’s not in line with any of the rest of `ByteBuffer`s methods. We don’t use Range anywhere else. 

More generally, it doesn’t seem to be a common Swift pattern for ranges to be arguments to anything but subscripts. Even `stride` doesn’t take a range.

Reply:
hmm, but for a view a Range makes a lot of sense, no? You surely want to write `viewBytes(range: 0..<10)` if you want 10 bytes, no? I mean we could make it a subscript if you want `subscript(viewBytes viewBytes: Range<Int>) -> ByteBufferView`

Reply:
If it was a subscript, sure. But otherwise it forces the math you’ve done in your test: `index..<(index + count)`.

Reply:
I thin I would use `viewBytes(at: Int, length: Int)`